modules/instance_template/variables.tf

```
@@ -164,3 +164,18 @@ variable "shielded_instance_config" {
     enable_integrity_monitoring = true
   }
 }
+
+###########################
+# Public IP
+###########################
+variable "public_ip" {
+  description = "Determines whether a public IP address is added to your VM instance."
+  type        = bool
+  default     = false
+}
+
+variable "access_config" {
+  description = "Access configurations, i.e. IPs via which the VM instance can be accessed via the Internet."
+  type        = map(string)
```

This should be a list of objects since it can be specified multiple times and has an expected structure.

```suggestion
  type        = list(object({ nat_ip=string, public_ptr_domain_name=string, network_tier=string }))
```

---

Yes, didn't want to make `nat_ip`, `public_ptr_domain_name`, `network_tier` required. You can specify a single `nat_ip` property as I showed in the example https://github.com/terraform-google-modules/terraform-google-vm/pull/43/files#diff-4a228bd286500151c6c190c61d93bf11R23-R25 and you're good to go.

---

now, since the properties are required, I have to specify them explicitly, e.g.
```
  access_config   = {
    nat_ip            = google_compute_address.ip_address.address
    network_tier = "PREMIUM"
  }
```
changed the example respectively.

---

This is expected behaviour of objects. There is a [request to support optional arguments](https://github.com/hashicorp/terraform/issues/19898), but in the mean time, you just need to specify `null` for arguments you want to omit.